### PR TITLE
Make objax.nn.Sequential more powerful.

### PR DESCRIPTION
**Pull request by david-berthelot**
- Support slicing to run part of the network.
- Support modules with arbitrary number of inputs/outputs.

### CODE REVIEW

File: docs/source/objax/nn.rst [hunk 1]
```diff
@@ -122,12 +122,18 @@ objax.nn
         ml = objax.nn.Sequential([objax.nn.Linear(2, 3), objax.functional.relu,
                                   objax.nn.Linear(3, 4)])
         x = objax.random.normal((10, 2))
-        y = ml(x)  # Runs all the operations (Linear -> ReLU -> Linear).
+        y = ml(x)       # Runs all the operations (Linear -> ReLU -> BatchNorm0D -> Linear).
```
Comment (AlexeyKurakin) [inline docs/source/objax/nn.rst:125]: I don't see `BatchNorm0D` in the definition of `ml` two lines above

File: objax/module.py [hunk 2]
```diff
@@ -74,6 +74,12 @@ def vars(self, scope: str = '') -> VarCollection:
                 vc.update(v.vars(scope=f'{scope}[{p}]'))
         return vc
 
+    def __getitem__(self, key: Union[int, slice]):
```
Comment (AlexeyKurakin) [inline objax/module.py:77]: Do we need docstring for RST documentation?

Comment (david-berthelot) [reply objax/module.py:77]: I don't think so. It's the standard list square bracket operator overloaded so it doesn't return a list but instead a ModuleList or Sequential. It's not meant to show up in the documentation either.

File: docs/source/objax/nn.rst (hunk 1 above)
Comment (david-berthelot) [reply docs/source/objax/nn.rst]: Fixed

**Review by AlexeyKurakin — APPROVED** [review]
